backend/lib/endpoints/posts.js

```diff
@@ -563,6 +578,44 @@ async function routes(app) {
     },
   );
 
+  //Update "Share"
+  app.put(
+    "/:postId/shares/:actorId",
+    {
+      preValidation: [app.authenticate, app.setActor],
+      schema: sharePostSchema,
+    },
+    async (req) => {
+      const {
+        actor,
+        userId,
+        params: { postId },
+      } = req;
+
+      const [updateErr, updatedPost] = await app.to(
+        Post.findOneAndUpdate(
+          { _id: postId },
+          { $push: { shares: actor._id } },
+          { new: true },
+        ),
+      );
+      if (updateErr) {
+        req.log.error(updateErr, "Failed sharing post");
+        throw app.httpErrors.internalServerError();
+      } else if (updatedPost === null) {
+        throw app.httpErrors.notFound();
+      }
+
+      // action, post, actorId (triggredBy), authUserId, details
+      app.notifier.notify("shares", updatedPost, actor._id, userId);
```

Comment:
this is duplicate with the SocketIO logic, since the `shares` has been updated there, updating here another time could result in a duplicated `notification` is written to DB, and user may receive 2 email notifications because of this I assume.